azext_iot/central/models/template.py

```diff
@@ -15,30 +15,89 @@ def __init__(self, template: dict):
             self.name = template.get("displayName")
             self.interfaces = self._extract_interfaces(template)
             self.schema_names = self._extract_schema_names(self.interfaces)
+            self.components = self._extract_components(template)
```

Comment:
I think having all these mangled under the same class might introduce some very difficult to see bugs/create maintenance issues. But depending on time constraints I can see why we did this.

I see 3 options:

1) Keep as is
2) Split template parsing into 2 files (templateV1, templateV2), and have a factory that builds them appropriately
3) Split parsing of templates into individual components and have interface/component/etc level parsers.

I think doing (2) makes sense here, since that would also make deprecation of v1 templates super trivial.

I'll let you file backlog items/think about this further, but I won't block on it.

Reply:
option 3 is better since components are type of interfaces and v2 has additive components. Adding separate parsers will make sense if the content is completely different and is exclusive.  Backlog item exists to address this for Jan.